Remember my credit card feature with adyen (#430)
front-commerce/developers.front-commerce.com: source/docs/advanced/payments/adyen.md

- `FRONT_COMMERCE_ADYEN_NOTIFICATION_PASSWORD` (more on [webhooks below)](#Add-webhook) you create it. You then have to copy it to the webhook section in your Adyen Customer Area under `Developers > Webhooks > [your_prefered_webhook] > Authentication > Password`
- `FRONT_COMMERCE_ADYEN_HMAC_KEY` (more on [webhooks below](#Add-webhook)) create it from the webhook section in your Adyen Customer Area under `Developers > Webhooks > [your_prefered_webhook] > Additional Settings > HMAC key`
- `FRONT_COMMERCE_ADYEN_PREVIOUS_HMAC_KEY` (more on [webhooks below](#Add-webhook)) when you regenerate your HMAC key configure this to be the old one

> - `FRONT_COMMERCE_ADYEN_STORE_DETAILS` Boolean (more on [Adyen's Stored Payment Methods](#Enable-Adyen’s-Stored-Payment-Methods-optional-1)). If you want to enable Adyen's Stored Payment Methods (Tokenization).

Reviewer: I didn't realized on the merge request, but the variable name seems a bit weird, why not `FRONT_COMMERCE_ADYEN_STORE_PAYMENT_METHODS` ?

Author: I thought about this as well, but the feature isn't really about storing a payment method, but storing details of a specific one (in the case of Adyen, it's specific to credit cards), so I found it was a bit of a "faux-ami".

Also in the code (and calls), they mix `storeDetails` and `storedPaymentMethods` so that's a bit of bias on my end I guess.

Author: Should I make a quick MR to fix this?

Reviewer: It does not seem critical either, @paul-vd @real34, MR or not MR ?

Reviewer 2: I also agree with the sense of having this specify that it's related to payment so `FRONT_COMMERCE_ADYEN_STORE_PAYMENT_DETAILS`

> Should I make a quick MR to fix this?

@Kagaminara I think so, it's better to do it before we do a release to avoid a break change.

Author: Done [here](https://github.com/front-commerce/developers.front-commerce.com/pull/430)

- Additional Settings
  - HMAC Key (HEX Encoded): click generate and copy it to `FRONT_COMMERCE_ADYEN_HMAC_KEY` environment variable defined above

### Enable Adyen's Stored Payment Methods (optional)

> You can opt for Adyen to store your customers' credit card information to allow a faster checkout experience.
> 
> In order to do so, you just need to add `FRONT_COMMERCE_ADYEN_STORE_DETAILS=true` in your `.env` file.

Reviewer: ```suggestion
You can opt for Adyen to store your customers' credit card information and allow a faster checkout experienceby adding `FRONT_COMMERCE_ADYEN_STORE_DETAILS=true` in your `.env` file.
```

> That's it! If you're using the default Front-Commerce Adyen component, your customer should now be able to save their credit card information during checkout by checking the "Save for my next payment" checkbox.

Reviewer: There is already a "that's it" topic just below, this feels weird 
```suggestion
If you are using the default Front-Commerce Adyen component, your customer should now be able to save their credit card information during checkout by checking the "Save for my next payment" checkbox.
```

[Same as above](#Register-custom-styles-optional)

### Enable Adyen's Stored Payment Methods (optional)

> You can opt for Adyen to store your customers' credit cards informations to allow them a faster checkout experience.

Reviewer 2: ```suggestion
You can opt for Adyen to store your customers' credit card information to allow a faster checkout experience.
```

In order to do so:

- Add `FRONT_COMMERCE_ADYEN_STORE_DETAILS=true` in your `.env` file
- Follow [Adyen's guide](https://docs.adyen.com/plugins/magento-2/set-up-tokenization?tab=magento-vault-version-8-2-0_2#set-up-adyen-tokenization) on how to setup tokenization using Adyen's Magento module.

> That's it! If you're using the default Front-Commerce Adyen component, your customer should now be able to save their credit card information during checkout by checking the "Save for my next payment" checkbox.

Reviewer: Same here : there is already a "that's it" topic below
```suggestion
If you're using the default Front-Commerce Adyen component, your customer should now be able to save their credit card information during checkout by checking the "Save for my next payment" checkbox.
```

> That's it ! If you're using the default Front-Commerce Adyen component, your customer should now be able to save their credit card information during checkout by checking "Save for my next payment" checkbox.

Reviewer 2: ```suggestion
That's it! If you're using the default Front-Commerce Adyen component, your customer should now be able to save their credit card information during checkout by checking the "Save for my next payment" checkbox.
```
